lib/sorbet-rails/gem_plugins/attr_json_plugin.rb

```diff
@@ -0,0 +1,117 @@
+# typed: false
+class AttrJsonPlugin < SorbetRails::ModelPlugins::Base
+  sig { override.params(root: Parlour::RbiGenerator::Namespace).void }
+  def generate(root)
+    return unless @model_class.include?(::AttrJson::Record)
+
+    # Module for instance methods
+    obj_custom_module_name = self.model_module_name("GeneratedAttrJsonMethods")
+    obj_custom_module_rbi = root.create_module(obj_custom_module_name)
+
+    # Module for class methods
+    klass_custom_module_name = self.model_module_name("GeneratedAttrJsonClassMethods")
+    klass_custom_module_rbi = root.create_module(klass_custom_module_name)
+
+    # here we re-create the model class!
+    model_class_rbi = root.create_class(self.model_class_name)
+    model_class_rbi.create_include(obj_custom_module_name)
+    model_class_rbi.create_extend(klass_custom_module_name)
+
+    # then create custom methods, constants, etc. for this module.
+    add_class_methods(klass_custom_module_rbi)
+
+    @model_class.attr_json_registry.definitions.each do |definition|
+      add_methods_for_attributes(definition, obj_custom_module_rbi)
+    end
+
+    if @model_class.include?(::AttrJson::Record::Dirty)
+      obj_custom_module_rbi.create_method(
+        'attr_json_changes',
+        returns: 'AttrJson::Record::Dirty::Implementation'
+      )
+    end
+  end
+
+  private
+
+  sig { params(custom_module_rbi: Parlour::RbiGenerator::ModuleNamespace).void }
+  def add_class_methods(custom_module_rbi)
+    custom_module_rbi.create_method(
+      'attr_json_config',
+      parameters: [
+        ::Parlour::RbiGenerator::Parameter.new(
+          'new_values',
+          type: 'T.nilable(T::Hash[Symbol, T.untyped])'
+        )
+      ],
+      returns: 'T::Hash[Symbol, T.untyped]'
+    )
+
+    custom_module_rbi.create_method(
+      'attr_json',
+      parameters: [
+        ::Parlour::RbiGenerator::Parameter.new(
+          'name',
+          type: 'T.any(Symbol, String)'
+        ),
+        ::Parlour::RbiGenerator::Parameter.new(
+          'type',
+          type: 'T.any(Symbol, ActiveModel::Type::Value)'
+        ),
+        ::Parlour::RbiGenerator::Parameter.new(
+          '**options',
+          type: 'T.untyped'
+        )
+      ]
+    )
+  end
+
+  sig do
+    params(
+      definition: ::AttrJson::AttributeDefinition,
+      custom_module_rbi: Parlour::RbiGenerator::ModuleNamespace
+    )
+    .void
+  end
+  def add_methods_for_attributes(definition, custom_module_rbi)
+    definition_type = get_definition_type(definition)
+
+    custom_module_rbi.create_method(
+        "#{definition.name}=",
+        parameters: [
+          ::Parlour::RbiGenerator::Parameter.new(
+            'value',
+            type: 'T.untyped'
+          )
+        ],
+        returns: 'T.untyped'
+    )
+
+    custom_module_rbi.create_method(definition.name.to_s,
+                                    returns: definition_type)
```

Comment:
nit: to keep the styling consistent, can you update the style following this suggestion:
```suggestion
    custom_module_rbi.create_method(
      definition.name.to_s,
      returns: definition_type,
    )
```